src/message.rs

```diff
@@ -146,50 +133,105 @@ impl KafkaMessageSender {
         list.clear();
     }
 
-    fn flush(&mut self) {
+    fn flush(&self) {
         self.flush_list(BALANCES_TOPIC);
         self.flush_list(ORDERS_TOPIC);
-        self.flush_list(DEALS_TOPIC);
+        self.flush_list(TRADES_TOPIC);
         self.producer.poll(Duration::from_millis(0));
     }
 
-    pub fn start_timer(&self) {
-        let mut ticker = tokio::time::interval(std::time::Duration::from_millis(100));
-        let self_ptr: *mut Self = self as *const Self as *mut Self;
-        tokio::task::spawn_local(async move {
-            loop {
-                ticker.tick().await;
-                unsafe {
-                    (*self_ptr).flush();
+    pub fn start(self) {
+        let mut last_flush_time = Instant::now();
+        let flush_interval = std::time::Duration::from_millis(100);
+        let timeout_interval = std::time::Duration::from_millis(100);
+        loop {
+            if self.is_block() {
+                // skip receiving from channel, so main server can know something goes wrong
+                // sleep to avoid cpu 100% usage
+                thread::sleep(flush_interval);
+            } else {
+                match self.receiver.recv_timeout(timeout_interval) {
+                    Ok((topic, message)) => {
```

Comment:
should we also add logging here?

Reply:
Ok(xx) and Timeout are both normal paths. So log here is not needed.